doc/semigroups.xml

```diff
@@ -95,14 +95,40 @@ gap> GeneratorsOfSemigroup(S);
 <#GAPDoc Label="ClosureSemigroup">
   <ManSection>
     <Oper Name = "ClosureSemigroup" Arg = "S, coll [, opts]"/>
-    <Returns>A semigroup or monoid.</Returns>
+    <Oper Name = "ClosureMonoid" Arg = "S, coll [, opts]"/>
+    <Oper Name = "ClosureInverseSemigroup" Arg = "S, coll [, opts]"/>
+    <Oper Name = "ClosureInverseMonoid" Arg = "S, coll [, opts]"/>
+    <Returns>A semigroup, monoid, inverse semigroup, or inverse
+      monoid.</Returns>
     <Description>
-      This function returns the semigroup or monoid generated by the
-      semigroup <A>S</A> and the collection of elements 
-      <A>coll</A> after removing duplicates and elements
-      from <A>coll</A> that are already in <A>S</A>. In most cases, the new
+      These operations return the semigroup, monoid, inverse semigroup or
+      inverse monoid generated by the argument <A>S</A> and the collection of
+      elements <A>coll</A> after removing duplicates and elements from
+      <A>coll</A> that are already in <A>S</A>. In most cases, the new
       semigroup knows at least as much information about its structure as was
       already known about that of <A>S</A>. <P/>
+      
+      When <C>X</C> is any of
+      <Ref Func="Semigroup" BookName="ref"/>,
+      <Ref Func="Monoid" BookName="ref"/>,
+      <Ref Func="InverseSemigroup" BookName="ref"/>, or
+      <Ref Func="InverseMonoid" BookName="ref"/>, 
+      the argument <A>S</A> of the operation <C>ClosureX</C> must belong to the
+      category <C>IsX</C>, and <C>ClosureX(<A>S</A>, <A>coll</A>)</C> returns
+      an object in the category <C>IsX</C> such that 
+      <Log>
+        ClosureX(S, coll) = X(S, coll);</Log>
+      but may have fewer generators, if for example,
+      <A>coll</A> contains a duplicates or elements already known to belong to
+      <A>S</A>.<P/>
+      
+      For example,    the argument <A>S</A> of <C>ClosureInverseSemigroup</C>
+      must be an inverse semigroup in the category <Ref
+        Filt="IsInverseSemigroup" BookName="ref"/> and an inverse semigroup is
```

Comment:
This paragraph mentions twice that an inverse semigroup is returned.  I'd scrap the first one.

Also there's some funny whitespace in this parapraph.